react-hot-button-component/src/hot-button.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+
+export default class HotButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.state = { clicks: 0 };
+  }
+
+  handleClick() {
+    this.setState({
+      clicks: this.state.clicks + 1
+    });
+  }
+
+  render() {
+    const timesClicked = this.state.clicks;
+    const text = 'Hot Button';
+    if (timesClicked < 3) {
+      return (
+        <button className='initial' onClick={this.handleClick}>{text}</button>
+      );
+    } else if (timesClicked < 6) {
+      return (
+        <button className='click3' onClick={this.handleClick}>{text}</button>
+      );
+    } else if (timesClicked < 9) {
+      return (
+        <button className='click6' onClick={this.handleClick}>{text}</button>
+      );
+    } else if (timesClicked < 12) {
+      return (
+        <button className='click9' onClick={this.handleClick}>{text}</button>
+      );
+    } else if (timesClicked < 15) {
+      return (
+        <button className='click12' onClick={this.handleClick}>{text}</button>
+      );
+    } else if (timesClicked < 18) {
+      return (
+        <button className='click15' onClick={this.handleClick}>{text}</button>
+      );
+    } else if (timesClicked >= 18) {
+      return (
+        <button className='click18' onClick={this.handleClick}>{text}</button>
+      );
+    }
+  }
```

Comment:
This is repetitive and would be hard to maintain. A better approach would be to use the if statements to determine the correct class to apply, then return a single button at the end.
```JSX
const timesClicked = this.state.clicks;
const text = 'Hot Button';
let btnClass = '';

if (timesClicked < 3) {
  btnClass = 'initial';
} else if (timesClicked < 6) {
  btnClass = 'click3';
} else if (timesClicked < 9) {
  btnClass = 'click6';
} else if (timesClicked < 12) {
  btnClass = 'click9';
} else if (timesClicked < 15) {
  btnClass = 'click12';
} else if (timesClicked < 18) {
  btnClass = 'click15';
} else if (timesClicked >= 18) {
  btnClass = 'click18';
}

return (
  <button className={btnClass} onClick={this.handleClick}>{text}</button>
);
```